### ISSUE TITLE
Fix Graph Validator

### ISSUE THREAD

**Issue by mshadbolt** [ebi-ait/hca-ebi-wrangler-central#179]
The graph validator has a few issues.

1. Linking doesn't seem correct since the change was made to link all entities to project. This may have to do with the `direct_links` attribute of the imported xls

2. The functionality to load a submission from ingest doesn't seem to work, it loads 0 nodes and 0 edges

3. Some users encounter docker issues when reinstalling with the error: 
```
C02YV4HKLVDL:ingest-graph-validator mshadbolt$ ingest-graph-validator hydrate xls graph_example_sheets/WongAdultRetina-20190913.xlsx 
20-11-24 11:27:25 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
20-11-24 11:27:25 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
20-11-24 11:27:25 [ingest_graph_validator.ingest_graph_validator] - ERROR: no backend container found
```

**Comment by clairerye** [comment]
We also need to deploy the latest changes in pypi. @aaclan-ebi was adding maintainer/owner access to team members to make this easier to do in the future.
 https://pypi.org/project/ingest-graph-validator/

**Comment by ESapenaVentura** [comment]
The ticket + claire's comments do summarise pretty well the problems we discussed this morning in the operation review meeting.

Please correct me if I'm wrong but our main focus should be:

Development tasks:
- Create a ticket to investigate linking issue related to code that may have to do with ingest's `direct links`.
- Create a ticket to fix the ingest hydrator

Operational tasks:
- Create a ticket to write an SOP on how to deploy new versions of ingest-graph-validator to pipy ( @aaclan-ebi ?)

Am I missing anything @aaclan-ebi @mshadbolt @ami-day @rays22 ? Happy to create the tickets, but I need to know who to assign to each of them

**Comment by rays22** [comment]
> The ticket + claire's comments do summarise pretty well the problems we discussed this morning in the operation review meeting.
> 
> Please correct me if I'm wrong but our main focus should be:
> 
> Development tasks:
> 
> * Create a ticket to investigate linking issue related to code that may have to do with ingest's `direct links`.
> * Create a ticket to fix the ingest hydrator
> * Investigate how to fix the docker issues (Either removing docker from the equation or fixing the underlying problems)
> 
> Operational tasks:
> 
> * Create a ticket to write an SOP on how to deploy new versions of ingest-graph-validator to pipy ( @aaclan-ebi ?)
> 
> Am I missing anything @aaclan-ebi @mshadbolt @ami-day @rays22 ? Happy to create the tickets, but I need to know who to assign to each of them

The task list looks comprehensive to me. I am happy to do testing when we get to that point in any of the tasks.

**Comment by prabh-t** [comment]
1. _since the change was made to link all entities to project_ this change has been done to the core model but the migration hasn't happened yet. @aaclan-ebi @jacobwindsor since we have deployed the change without migration, could this be causing this issue?

2. this seems to be working for some submissions but not for others. for e.g.

this one loads fine
```
$ ingest-graph-validator hydrate ingest d5410c6e-612d-421a-a66f-2de5e04dd050
20-12-02 23:50:04 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
20-12-02 23:50:04 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
20-12-02 23:50:04 [ingest_graph_validator.hydrators.hydrator] - INFO: started ingest hydrator for subid [d5410c6e-612d-421a-a66f-2de5e04dd050]
20-12-02 23:50:12 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 191 nodes
20-12-02 23:50:12 [ingest_graph_validator.utils] - INFO: [get_nodes] took [15] ms
20-12-02 23:51:38 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 703 edges
20-12-02 23:51:38 [ingest_graph_validator.utils] - INFO: [get_edges] took [85603] ms
20-12-02 23:51:41 [ingest_graph_validator.hydrators.hydrator] - INFO: hydration finished
```

but not this one
```
$ ingest-graph-validator hydrate ingest 24435718-d77a-43df-a905-a11db5d22e46
20-12-02 23:49:12 [ingest_graph_validator.ingest_graph_validator] - INFO: starting neo4j docker instance
20-12-02 23:49:12 [ingest_graph_validator.ingest_graph_validator] - INFO: attached to backend container [neo4j-server]
20-12-02 23:49:13 [ingest_graph_validator.hydrators.hydrator] - INFO: started ingest hydrator for subid [24435718-d77a-43df-a905-a11db5d22e46]
20-12-02 23:49:14 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 0 nodes
20-12-02 23:49:14 [ingest_graph_validator.utils] - INFO: [get_nodes] took [0] ms
20-12-02 23:49:14 [ingest_graph_validator.hydrators.hydrator] - INFO: imported 0 edges
20-12-02 23:49:14 [ingest_graph_validator.utils] - INFO: [get_edges] took [0] ms
20-12-02 23:49:14 [ingest_graph_validator.hydrators.hydrator] - INFO: hydration finished
```

I am trying to determine the difference between these 2 submissions, it could be a linking issue and related to 1.

3. This was due to a bug which is now fixed and published to PyPhy (v0.6.1). Please upgrade to this version. I strongly recommend installing into a virtualenv.

```
$ virtualenv venv-test
$ . venv-test/bin/activate
$ pip install ingest-graph-validator
$ ingest-graph-validator init
$ ingest-graph-validator hydrate xls <path/to/xls>
$ ingest-graph-validator hydrate ingest <sub_uuid>
$ ingest-graph-validator action test <path/to/tests_dir>
$ ingest-graph-validator shutdown
```
